File: TUnit.TestProject/ClassConstructorTest.cs
using TUnit.TestProject.Dummy;

namespace TUnit.TestProject;

[ClassConstructor<DependencyInjectionClassConstructor>]
public class ClassConstructorTest(SomeAsyncDisposableClass someAsyncDisposableClass)
{
    [Test]
    public void Test()
    {
    }
}

File: TUnit.TestProject.FSharp/ClassConstructorTest.fs
namespace TUnit.TestProject.FSharp

open TUnit.Core

[<ClassConstructor(typeof<DependencyInjectionClassConstructor>)>]
type ClassConstructorTest(dummyReferenceTypeClass: DummyReferenceTypeClass) =

    member _.DummyReferenceTypeClass = dummyReferenceTypeClass

    [<Test>]
    member _.Test() = ()

